cnd/src/main.rs

```diff
@@ -43,6 +43,9 @@ fn main() -> anyhow::Result<()> {
 
     let settings = read_config(&options).and_then(Settings::from_config_file_and_defaults)?;
 
+    settings.bitcoin.bitcoind.verify_connection_details();
```

Comment:
What is the point of the `Sanity` trait abstraction, if you are not making use of it? (By that I mean, there is not function that abstracts over `Sanity` as in: `fn verify<C: Sanity>(connection: C)`)

These could just be functions :)

But I'd suggest a **less** object-oriented design where you don't delegate to the data but have separate functions here in `main` that allow you to pass the connection details of each ledger connector and it verify the connection:

```rust
async fn verify_bitcoind_connection(network: Network, url: Url) -> Result<...> {
   // do bitcoind verification
}

async fn verify_parity_connection(chain_id: u16, url: Url) -> Result<...> {
   // do parity verification
}
```

and then just call them sequentially:

```rust
verify_bitcoind_connection(settings.bitcoin.network, settings.bitcoin.connectors.bitcoind.node_url).await?;
verify_parity_connection(settings.ethereum.network, settings.ethereum.connectors.parity.node_url).await?;
```